x-pack/plugins/maps/server/data_indexing/indexing_routes.ts

```diff
@@ -101,4 +103,32 @@ export function initIndexingRoutes({
       }
     }
   );
+
+  router.get(
+    {
+      path: `${GET_MATCHING_INDEXES_PATH}/{indexPattern}`,
+      validate: {
+        params: schema.object({
+          indexPattern: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const result = await getMatchingIndexes(
+        request.params.indexPattern,
+        context.core.elasticsearch.client
+      );
+      if (result.success) {
+        return response.ok({ body: result });
+      } else {
+        if (result.error) {
+          logger.error(result.error);
+        }
+        return response.custom({
```

Comment:
This does not align with how the route is getting called in public. In public, you are casting the response to `MatchingIndexesResp` but here the route is returning an error message and not MatchingIndexesResp. I would recommend getting ride of all of the logic in this route and just return the results of getMatchingIndexes since it all ready handles catching any errors. You can just log the error in getMatchingIndexes.